app/services/duplicate_canonical_profile_resolver.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+# Class that deduplicates canonical profiles with identical benchmark/ref IDs
+class DuplicateCanonicalProfileResolver
+  class << self
+    def run!
+      duplicates.each do |duplicate|
+        profiles = Profile.canonical.includes(:child_profiles).where(
+          duplicate.slice(:ref_id, :benchmark_id)
+        ).order(created_at: :asc).offset(1)
+
+        next unless profiles.all? { |p| p.child_profiles.count.zero? }
```

Comment:
Just to say it out loud: our strategy for now is to do nothing in the case of child profiles, assuming we have none currently in prod. If we do, however, find some duplicate canonical profiles that do have children, we should probably log something so we know we have more work to do.

Reply:
How should I do logging from a migration? :thinking:

Reply:
You should be able to use the rails logger like normal: `Rails.logger.error(...)` It should go to stdout in our migrations.